rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+
+AvatarProps (badge, image, label, icon), BadgeProps (icon), ButtonProps (icon), ...
+
+## Slots and HTML attributes
+
+> Only Accordian and Menu components use ComponentProps<> with slots currently.
+
+AccordionSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+AccordionHeaderSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+  button: ARIAButtonShorthandProps;
+  expandIcon: AccordionHeaderExpandIconProps;
+  icon?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+  children: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+AccordianItemSlots, AccordianPanelSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuSlots
+
+```ts
+(empty type)
+```
+
+MenuDividerSlots, MenuGroupSlots, MenuGroupHeaderSlots, MenuListSlots, MenuPopoverSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuItemSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+icon?: React.HTMLAttributes<HTMLElement>;
+checkmark?: React.HTMLAttributes<HTMLElement>;
+submenuIndicator?: React.HTMLAttributes<HTMLElement>;
+content: React.HTMLAttributes<HTMLElement>;
+secondaryContent?: React.HTMLAttributes<HTMLElement>;
+```
+
+## Compat and HTML attributes
+
+> The components not yet using slots extend ComponentPropsCompat and provide HTML attributes by extending the associated React HTML attribute types (e.g `React.HTMLAttributes<HTMLElement>`).
+> PortalProps and PopoverTriggerProps do not extend any HTML attributes, but provide children as a property directly in their types.
+
+AvatarProps,BadgeProps,DividerProps, TooltipProps
+
+```ts
+extends ComponentPropsCompat, React.HTMLAttributes<HTMLElement>
+```
+
+ButtonProps
+
+```ts
+ButtonProps = ComponentPropsCompat & React.ButtonHTMLAttributes<HTMLElement> & {...}
+```
+
+MenuButtonProps/MenuTriggerChildProps
+
+```ts
+MenuButtonProps = Omit<ButtonProps, 'iconPosition'>  & Partial<MenuTriggerChildProps> & {}
+
+MenuTriggerChildProps extends Required<Pick<React.HTMLAttributes<HTMLElement>,
+      | 'onClick'
+      | 'onMouseEnter'
+      | 'onMouseLeave'
+      | 'onContextMenu'
+      | 'onKeyDown'
+      | 'aria-haspopup'
+      | 'aria-expanded'
+      | 'id'
+    >
+```
+
+ImageProps
+
+```ts
+extends ComponentPropsCompat,
+React.ImgHTMLAttributes<HTMLImageElement>
+```
+
+LabelProps
+
+```ts
+extends ComponentPropsCompat,
+React.LabelHTMLAttributes<HTMLLabelElement>
+```
+
+LinkProps
+
+```ts
+LinkProps = ComponentPropsCompat &
+  React.AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement> &
+  Omit<React.ButtonHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement>, 'type'> & {...}
+```
+
+PopoverProps
+
+```ts
+// PortalProps is an interface with children and mountNode.
```

Comment:
This is intentional as `Portal` does render a DOM node, but it's not rendered as a part of the render cycle and shouldn't be exposed publicly